ocaml/xapi/xapi_vdi.ml

```diff
@@ -238,7 +241,8 @@ let check_operation_error ~__context ?(sr_records=[]) ?(pbd_records=[]) ?(vbd_re
         then Some (Api_errors.vdi_on_boot_mode_incompatible_with_operation, [])
         else None
       | `mirror | `clone | `generate_config | `force_unlock | `set_on_boot | `export_changed_blocks | `blocked | `update -> None
-      end
+      | `unknown -> None (* Note, this is caught above *)
```

Comment:
Really minor comment, but I think just in case, we could return an error here, to make the code more robust and protect against unexpected code changes:
```ocaml
| `unknown -> (Api_errors.sr_operation_not_supported, ["Unknown operation"]) (* Note, this is caught above *)
```

Reply:
Yes, good idea - defense in depth.

Actually, what this is missing is some unit tests (which would catch the sort of error this defensive style would be trying to prevent). I'll write some.